### PR TITLE
Finish missing sources from Datastream Stream

### PR DESCRIPTION
**Pull request by iperetz-goo**
Add missing fields and behavior from recently implemented Datastream Stream resource [PR link](https://github.com/GoogleCloudPlatform/magic-modules/pull/6806)

Changes:
1. Add Postgres & Oracle source configs
2. Call an additional update after create stream if the desired_state was RUNNING.
3. Add support for validation results in operation errors

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:REPLACEME

```

### CODE REVIEW

File: mmv1/templates/terraform/post_create/datastream_stream.go.erb [hunk 1]
```diff
@@ -14,4 +14,11 @@
 -%>
 if err := waitForDatastreamStreamReady(d, config, d.Timeout(schema.TimeoutCreate) - time.Minute); err != nil {
   return fmt.Errorf("Error waiting for Stream %q to be NOT_STARTED or RUNNING during creation: %q", d.Get("name").(string), err)
+}
+
+if d.Get("state") != d.Get("desired_state") {
```
Comment (c2thorn) [inline mmv1/templates/terraform/post_create/datastream_stream.go.erb:19]: This seems to be completely unrelated to adding the other fields. Is the purpose of doing another Update call just a retry of sorts? Have you tested that this is effective?

File: scripts/make-in-container.sh [hunk 2]
```diff
@@ -46,4 +46,4 @@ build_dev_container() {
         -t "${DEV_IMAGE}"
 }
 
-main "$@"
\ No newline at end of file
+main "$@"
```
Comment (c2thorn) [inline scripts/make-in-container.sh:49]: Even though it is trivial, can we remove this change to not affect the file history?

**Review by c2thorn — CHANGES_REQUESTED** [review]
Apologies for the delay. I have a couple of questions. I have not run tests related to this PR yet, since I think they can be run in the automatic VCR tests instead of being skipped. Let me know

File: mmv1/products/datastream/terraform.yaml [hunk 3]
```diff
@@ -163,6 +175,27 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           destination_connection_profile_id: "destination-profile"
         test_vars_overrides:
           deletion_protection: "false"
+      - !ruby/object:Provider::Terraform::Examples
+        name: "datastream_stream_postgresql"
+        pull_external: true
+        primary_resource_id: "default"
+        # Random provider
+        skip_vcr: true
```
[the remark this answers is not in the document]
Comment (iperetz-goo) [reply mmv1/products/datastream/terraform.yaml:183]: done

File: mmv1/templates/terraform/post_create/datastream_stream.go.erb (hunk 1 above)
Comment (iperetz-goo) [reply mmv1/templates/terraform/post_create/datastream_stream.go.erb:19]: This is a bug fix, datastream requires an additional update call when desired_state=RUNNING

File: scripts/make-in-container.sh (hunk 2 above)
Comment (iperetz-goo) [reply scripts/make-in-container.sh:49]: done

**Comment by roaks3** [comment]
FYI @c2thorn this was accidentally closed, and the new PR is https://github.com/GoogleCloudPlatform/magic-modules/pull/7175
